web_generate_assets/hooks.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+import logging
+import os
+
+from odoo import fields, registry
+
+logger = logging.getLogger(__name__)
+
+
+def active_cron_assets():
+    """Plan the next execution of the cron responsible to generate assets."""
+    if os.environ.get("RUNNING_ENV") == "dev":
+        return
+    dbname = os.environ.get("DB_NAME")
+    reg = registry(dbname)
+    with reg.cursor() as cr:
+        cron_module, cron_ref = "web_generate_assets", "cron_generate_assets"
+        query = """
+           SELECT model, res_id
+           FROM ir_model_data
+           WHERE module=%s
+           AND name=%s;
+        """
+        args = (cron_module, cron_ref)
+        cr.execute(query, args)
+        row = cr.fetchone()
+        # post_load hook is called before the update of the module so the
+        # ir_cron record doesn't exist on first install
+        if row:
+            model, res_id = row
+            if model != "ir.cron":
+                return
```

Comment:
```suggestion
import hashlib
import logging
import os
import struct

from odoo import fields, registry

logger = logging.getLogger(__name__)


def active_cron_assets():
    """Plan the next execution of the cron responsible to generate assets."""
    if os.environ.get("RUNNING_ENV") == "dev":
        return
    dbname = os.environ.get("DB_NAME")
    reg = registry(dbname)
    with reg.cursor() as cr:
        cron_module, cron_ref = "web_generate_assets", "cron_generate_assets"
        query = """
           SELECT model, res_id
           FROM ir_model_data
           WHERE module=%s
           AND name=%s;
        """
        args = (cron_module, cron_ref)
        cr.execute(query, args)
        row = cr.fetchone()
        # post_load hook is called before the update of the module so the
        # ir_cron record doesn't exist on first install
        if row:
            model, res_id = row
            if model != "ir.cron":
                return
            hasher = hashlib.sha1(str("web_generate_assets").encode())
            int_lock = struct.unpack("q", hasher.digest()[:8])
            cr.execute("SELECT pg_advisory_xact_lock(%s);", (int_lock,))
```

Reply:
Since we could have dedicated odoo services for web or cronjobs, we should create a lock. Otherwise we could get a serialize access error.